### PR TITLE
`get_wavelengths_from_dv`: handle `dv.extFloats` that are `np.recarray`s

### PR DESCRIPTION
**Pull request by thomasmfish**
I've been testing on DV files from Cockpit, which doesn't write "proper" DV files. When testing on a "proper" DV file, `mrc.Mrc` handles things differently, storing the data as a 1D `np.recarray` of `np.record`s, with each record being the metadata for an image plane. I have adjusted `get_wavelengths_from_dv` to be able to handle either type efficiently.

### CODE REVIEW

**Review by thomasmfish — COMMENTED** [review]
@iandobbie this is the change that should allow proper OMX files to work

File: src/sim_recon/images/dv.py [hunk 1]
```diff
@@ -75,25 +75,68 @@ def get_dv_axis_sizes(dv: mrc.Mrc) -> dict[str, int]:
...
+        # check each defined wavelength
+        ext_floats = ext_floats.reshape(*header_shape)
+
+        for c in range(header_shape[wavelength_index]):
+            indexes = [0, 0, 0]
+            indexes[wavelength_index] = c
+            yield Wavelengths(
+                # stored as a recarray of records, so getting the records need
+                # to be indexed separately
+                ext_floats[*indexes]["exWavelen"],
```
Comment (PeterC-DLS) [inline src/sim_recon/images/dv.py:94]: Does this `[*indexes]` work? Note, using a list (`[indexes]`) triggers fancy indexing whereas using a tuple (`[tuple(indexes)]`) doesn't

Comment (thomasmfish) [reply src/sim_recon/images/dv.py:94]: Since Python 3.11 due to [PEP646](https://peps.python.org/pep-0646/#change-1-star-expressions-in-indexes)

Comment (thomasmfish) [reply src/sim_recon/images/dv.py:94]: I can change to using `tuple(indexes)` but the minimum Python version is set to 3.12 and I do think this is cleaner.

Comment (PeterC-DLS) [reply src/sim_recon/images/dv.py:94]: It's fine as is - I haven't used 3.11+ as yet...

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**